### PR TITLE
Feature/v3 v4 token decoding

### LINKED ISSUE #7: V3 and V4 tokens are not supported
Currently, only the V2 tokens are correctly decoded, support should be added for V3 and V4.

### PR DESCRIPTION
**Pull request by contested-space**
This PR adds support for V3 and V4 tokens.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,3 +119,48 @@ this library is doing.
...
+| Offset (bytes) | Size (bytes) | Description |
+| -------------- | ------------ | ----------- |
+| 0              | 8            | Expiration time in milliseconds (Unix timestamp). Must be uint64 big endian. |
+| 8              | 8            | Timestamp when the token was generated in milliseconds (Unix timestamp). Must be uint64 big endian. |
+| 16             | 4            | Operator Site ID (unused) |
+| 20             | 1            | Operator Type (unused) |
+| 21             | 4            | Operator Version (unused) |
+| 25             | 4            | Operator Key ID (unused) |
+| 29             | 4            | Site key ID. Must be uint32 big endian. |
+| 23             | N            | Identity Payload (encrypted) |
```
Comment (rkallos) [inline README.md]: `23` should probably be `33`

File: lib/ex_uid2/encryption.ex [hunk 2]
```diff
@@ -70,6 +73,39 @@ defmodule ExUid2.Encryption do
...
+    case encrypt_v3(uid2, keyring, master_key_id, site_key_id) do
+      {:ok, token} -> {:ok, :base64.encode(token)}
+      {:error, _} = error -> error
+    end
+  end
+
+  @spec encrypt_v4_token(Uid2.t(), Keyring.t(), non_neg_integer(), non_neg_integer()) :: {:ok, binary()} | {:error, any()}
+  def encrypt_v4_token(uid2, keyring, master_key_id, site_key_id) do
+    case encrypt_v3(uid2, keyring, master_key_id, site_key_id) do
+      {:ok, token} -> {:ok, :base64.encode(token, %{mode: :urlsafe})}
```
Comment (rkallos) [inline lib/ex_uid2/encryption.ex:101]: I take it the only difference between encrypting v3 and v4 tokens is the 'urlsafe' mode of base64-encoding?

**Review by rkallos — APPROVED** [review]

File: lib/ex_uid2/encryption.ex (hunk 2 above)
Comment (contested-space) [reply lib/ex_uid2/encryption.ex:101]: That's correct.

File: README.md (hunk 1 above)
Comment (contested-space) [reply README.md]: A very good catch that proves you're reading my doc carefully ❤️ .

**Pull request merged**